src/main.cpp

```diff
@@ -5496,8 +5536,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                         // Probably non-standard or insufficient fee/priority
                         LogPrint("mempool", "   removed orphan tx %s\n", orphanHash.ToString());
                         vEraseQueue.push_back(orphanHash);
-                        assert(recentRejects);
-                        recentRejects->insert(orphanHash);
+                        if (recentRejects) recentRejects->insert(orphanHash);  // should always be true
```

Comment:
if for some reason the recentRejects CRollingBloomFilter is temporarily unavailable, the code will degrade gracefully -- it will just rerequest a transaction.  That's why I chose to make this a "normal" check